packages/core/src/codecatalyst/activation.ts

```diff
@@ -37,6 +37,17 @@ export async function activate(ctx: ExtContext): Promise<void> {
 
     await authProvider.restore()
 
+    // if connection is shared with CW, check if CC scopes are expired
+    // set suppressNextReauthPrompt to true to prevent notification on startup
+    const conn = authProvider.activeConnection
+    if (conn && authProvider.getConnectionState(conn).shared) {
+        try {
+            authProvider.suppressNextReauthPrompt = true
+            await createClient(conn)
+            authProvider.suppressNextReauthPrompt = false
```

Comment:
setting globals is fragile and is a less obvious interface for the operation of the auth subsystem. can this be plumbed through `createClient` instead?

```suggestion
            await createClient(conn, { suppressReauthPrompt = true })
```